lib/regexp/regexp.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package regexp provides functions related to regular expressions.
+package regexp // import "go.starlark.net/lib/regexp"
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+// Module regexp is a Starlark module of functions related to regular expressions.
+// The module defines the following functions:
+//
+//     compile(pattern) - Compiles a pattern in RE2 syntax (https://github.com/google/re2/wiki/Syntax) to a value of type 'regexp'.
+//                        Each call to compile returns a distinct regexp value.
+//                        A regexp value can be used for matching using its matches, find, find_all and other methods.
+//
+//     regexp.find(src) - Returns a string holding the text of the leftmost match in the given string of the regular expression regexp.
+//                         The result is "" if there is no match or if the pattern successfully matches an empty string.
+//
+//     regexp.find_all(src, max) - Returns a new, mutable list of all successive matches of the regular expression regexp. An empty list indicates
+//                                 no match. If max > 0, at most max strings are returned. If max == 0,
+//                                 an empty list is returned. If max < 0, all strings are returned.
+//                                 The parameter max is optional: by default no limit is applied.
+//
+//     regexp.find_submatches(src) - Returns a new, mutable list of strings holding the text of the leftmost match of the regular expression regexp
+//                                   in the given string and the matches, if any, of its subexpressions. An empty list indicates
+//                                   no match.
+//
+//     regexp.matches(src) - Reports whether the given string contains any match of the regular expression regexp.
+//
```

Comment:
Have you given any thought to including variants of the find* functions that return indices instead of strings? I guess that kind of optimized API might not be worth its weight in the context of Starlark?